app/scripts/views/mixins/flow-events-mixin.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Populates the flow model and initialises flow event handlers.
+
+define(function (require, exports, module) {
+  'use strict';
+
+  const Flow = require('models/flow');
+
+  const TAB_KEY = 9;
+
+  module.exports = {
+    afterRender () {
+      this.flow = new Flow({
+        sentryMetrics: this.sentryMetrics,
+        window: this.window
+      });
+
+      const flowId = this.flow.get('flowId');
+      const flowBegin = this.flow.get('flowBegin');
+
+      this.metrics.setFlowEventMetadata({
+        flowBeginTime: flowBegin,
+        flowId: flowId
+      });
+    },
+
+    events: {
+      'click a': '_clickFlowEventsLink',
+      'click input': '_engageFlowEventsForm',
+      'input input': '_engageFlowEventsForm',
+      'keyup input': '_keyupFlowEventsInput',
+      'submit': '_submitFlowEventsForm'
+    },
+
+    _clickFlowEventsLink (event) {
+      if (event && event.target && event.target.id) {
+        // Each link on the sign-in form emits a different event,
+        // named according to its id.
+        this.logFlowEvent(event.target.id, this.viewName);
+      }
+    },
+
+    _engageFlowEventsForm () {
+      this.logFlowEventOnce('engage', this.viewName);
+    },
+
+    _keyupFlowEventsInput (event) {
+      if (event.which === TAB_KEY && ! event.metaKey && ! event.ctrlKey && ! event.altKey) {
```

Comment:
The background is that [reviously I was listening for the focus event, with some extra logic to deal with our autofocus weirdness and to try and ensure sane behaviour when the window was focused/blurred. You pointed out problems with that approach because in some contexts the view is hidden in an iframe. You suggested using a `keyup` handler instead.

My concern  is that we are interpreting system-level signals in the states of `metaKey`, `ctrlKey` and `altKey`. Could the behaviour of those keys be changed to mean something different on some systems? It seems like a robust approach.

I was wondering whether you thought it might be worthwhile to try and persevere with an approach based on the `focus` event instead.